src/object_async/hello_async.cpp

```diff
@@ -165,8 +168,11 @@ struct AsyncHelloWorker : Nan::AsyncWorker
 {
 
     using Base = Nan::AsyncWorker;
+    // Make this class noncopyable
+    AsyncHelloWorker(AsyncHelloWorker const&) = delete;
+    AsyncHelloWorker& operator=(AsyncHelloWorker const&) = delete;
 
-    AsyncHelloWorker(bool louder, std::string const& name,
+    AsyncHelloWorker(bool louder, const std::string* name,
```

Comment:
> Why by const ptr?

To avoid the possibility of a dangling reference if a temporary string was accidentally passed to `new AsyncHelloWorker{louder, name_copy, new Nan::Callback{callback}};` as simulated at https://github.com/mapbox/node-cpp-skel/commit/6be05eb56e35d1ce9341438b6fa474ad897d3b87. See https://github.com/mapbox/node-cpp-skel/issues/65 for details.

> The difference is now the ptr can be null - with references it can not.

Ugh. Good point. Hard to know the ideal solution here. In general I want to:

a. avoid a copy (allowing both a pointer and a ref do that)
b. do more to make sure the `new AsyncHelloWorker` constructor cannot be misused.